nixos/modules/services/web-apps/pr-tracker.nix

```diff
@@ -0,0 +1,315 @@
+        description = ''
+          Whether you want this service to manage a clone of the nixpkgs
+          repo in `cfg.dataDir`.
+        '';
+      };
+
+      interval = mkOption {
+        type = types.str;
+        example = "15min";
+
+        default = "30min";
+        description = ''
+          How often to fetch nixpkgs if `cfg.nixpkgsClone.managedByModule` is true.
+        '';
+      };
+    };
+
+    userAgent = mkOption {
+      type = types.str;
+      example = "my pr-tracker";
+
+      description = ''
+        The User-Agent string to use when contacting the GitHub API.
+      '';
+    };
+
+    sourceUrl = mkOption {
+      type = types.str;
+      example = "https://github.com/me/my-pr-tracker-fork";
+
+      default = "https://git.qyliss.net/pr-tracker";
+      description = ''
+        The URL where users can download the program's source code.
+      '';
+    };
+
+    mountPath = mkOption {
+      type = with types; nullOr str;
+      example = "pr-tracker.html";
+
+      default = null;
+      description = ''
+        A "mount" path can be specified, which will be prefixed to all
+        of the server's routes, so that it can be served at a non-root
+        HTTP path.
+      '';
+    };
+
+    user = mkOption {
+      default = "pr-tracker";
+      type = types.str;
+      description = ''
+        User account under which pr-tracker runs.
+
+        ::: {.note}
+        If left as the default value this user will automatically be created
+        on system activation, otherwise you are responsible for ensuring the
+        user exists before the pr-tracker service starts.
+        :::
+      '';
+    };
+
+    group = mkOption {
+      default = "pr-tracker";
+      type = types.str;
+      description = ''
+        Group account under which pr-tracker runs.
+
+        ::: {.note}
+        If left as the default value this user will automatically be created
+        on system activation, otherwise you are responsible for ensuring the
+        user exists before the pr-tracker service starts.
+        :::
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      example = "/home/nixos/git";
+
+      default = "/var/lib/pr-tracker";
+      description = ''
+        The data directory for pr-tracker.
+
+        ::: {.note}
+        If left as the default value this directory will automatically be created
+        before the pr-tracker server starts, otherwise you are responsible for ensuring
+        the directory exists with appropriate ownership and permissions.
+        :::
+      '';
+    };
+
+    listen = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      example = "127.0.0.1";
+      description = ''
+        The IP address to bind the socket to.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      description = ''
+        TCP port where pr-tracker socket listens.
+      '';
+    };
+
+    openFirewall = mkOption {
```

Comment:
I don't think you would want to expose this service as is to the web. Since the nginx config is a bit none standard I would suggest to put it into this module.

I use the following

```
        "tracker.example.de" = {
          forceSSL = true;
          enableACME = true;
          locations."/" = {
            proxyPass = "http://unix:/run/pr-tracker/pr-tracker.sock:/pr-tracker.html";
            extraConfig = ''
              proxy_http_version 1.1;
            '';
          };
        };
```

Reply:
I don't think setting up nginx would be best. I personally don't use it, I use caddy instead.

Maybe we could set the default listen IP to `127.0.0.1` instead?

Reply:
nginx is the standard proxy we use in nginx. Lets make an option and everyone who wants to tinker with something else can do that but everyone who just wants to expose it, does not need to hand copy that very specific proxyPass. First time I set it up, it took me like 5 tries and I know what I am doing with nginx ;)

Why should we listen on a port which is likely already taken if we can listen on a socket? Sockets are usually way easier to handle and more compatible.

Reply:
Do you have any examples of options like this elsewhere in nixpkgs?